src/components/Routes/Recovery.tsx

```diff
@@ -1 +1,98 @@
-// This feature is not yet supported in React Native
+// This file renders the recovery screen.
+import {
+  SelfServiceBrowserLocationChangeRequiredError,
+  SelfServiceRecoveryFlow,
+  SubmitSelfServiceRecoveryFlowBody
+} from '@ory/client'
+import { useFocusEffect } from '@react-navigation/native'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useContext, useState } from 'react'
+import { handleFormSubmitError } from '../../helpers/form'
+import { newKratosSdk } from '../../helpers/sdk'
+import { AuthContext } from '../AuthProvider'
+import AuthLayout from '../Layout/AuthLayout'
+import { RootStackParamList } from '../Navigation'
+import { SelfServiceFlow } from '../Ory/Ui'
+import { ProjectContext } from '../ProjectProvider'
+import AuthSubTitle from '../Styled/AuthSubTitle'
+import StyledCard from '../Styled/StyledCard'
+
+type Props = StackScreenProps<RootStackParamList, 'Recovery'>
+
+const Recovery = ({ navigation }: Props) => {
+  const [flow, setFlow] = useState<SelfServiceRecoveryFlow | undefined>(
+    undefined
+  )
+  const { project } = useContext(ProjectContext)
+  const { setSession } = useContext(AuthContext)
+
+  const initializeFlow = () =>
+    newKratosSdk(project)
+      .initializeSelfServiceRecoveryFlowWithoutBrowser()
+      .then((response) => {
+        const { data: flow } = response
+        // The flow was initialized successfully, let's set the form data:
+        setFlow(flow)
+      })
+      .catch(console.error)
+
+  // When the component is mounted, we initialize a new recovery flow:
+  useFocusEffect(
+    React.useCallback(() => {
+      initializeFlow()
+
+      return () => {
+        setFlow(undefined)
+      }
+    }, [project])
+  )
+
+  const handleRedirection = async (err: SelfServiceBrowserLocationChangeRequiredError) => {
+    // A SelfServiceBrowserLocationChangeRequiredError indicates the `success` of the recovery api flow
+    // as after a recovery the user _needs_ to update their password in a settings flow
```

Comment:
This is a bit strange right? We have a success state but receive an error and need to handle the error. I know that we discussed it like this, but looking at this in its pure state I have to say that it feels very weird...

Reply:
It is. 

Also applies to the SPA flows, I guess. Maybe we could return the settings flow + sc 200 instead? 

Or do a proper HTTP redirect (3xx) to `/selfservice/flows/settings?flow_id=xxx` and then add some logic that detects the type of flow returned in the success?